### PR TITLE
Alerting: Filter folders based on users' permissions

### LINKED ISSUE #49672: Apply access control to the folder picker

### PR DESCRIPTION
**Pull request by konrad147**
**What this PR does / why we need it**:
This PR adds users' permission validation to the `RuleFolderPicker` component

**Which issue(s) this PR fixes**:
Fixes #49672

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/Select/FolderPicker.tsx [hunk 1]
```diff
@@ -183,6 +196,7 @@ export class FolderPicker extends PureComponent<Props, State> {
       <div data-testid={selectors.components.FolderPicker.containerV2}>
         <AsyncSelect
           inputId={inputId}
+          isOpen={true}
```
Comment (gillesdemey) [inline public/app/core/components/Select/FolderPicker.tsx:199]: Any particular reason why we had to add this prop here?

File: public/app/features/alerting/unified/components/rule-editor/DetailsStep.tsx [hunk 2]
```diff
@@ -21,7 +21,11 @@ const recordingRuleNameValidationPattern = {
   value: /^[a-zA-Z_:][a-zA-Z0-9_:]*$/,
 };
 
-export const DetailsStep: FC = () => {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rule-editor/DetailsStep.tsx:24]: We could have specified `FC<DetailsStepProps>` but technically this one doesn't render any `children` from props so it's 👌

File: public/app/features/alerting/unified/hooks/useIsRuleEditable.ts [hunk 3]
```diff
@@ -29,24 +27,25 @@ export function useIsRuleEditable(rulesSourceName: string, rule?: RulerRuleDTO):
...
     if (!folderUID) {
       throw new Error(
         `Rule ${rule.grafana_alert.title} does not have a folder uid, cannot determine if it is editable.`
       );
     }
     return {
-      isEditable: hasEditPermission && folder?.canSave,
-      isRemovable: hasRemovePermission && folder?.canSave,
+      isEditable: contextSrv.hasAccess(rulePermission.update, folder?.canSave ?? false),
+      isRemovable: contextSrv.hasAccess(rulePermission.delete, folder?.canSave ?? false),
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/hooks/useIsRuleEditable.ts:39]: nit: Can we move this back to separate variables like `hasEditPermission` and `hasRemovePermission`? It always makes it a bit easier to read the intentions and business logic of the code.

**Review by gillesdemey — APPROVED** [review]
Great work! Just a few small nits (non-blocking) 🚀

File: public/app/core/components/Select/FolderPicker.tsx (hunk 1 above)
Comment (konrad147) [reply public/app/core/components/Select/FolderPicker.tsx:199]: I was testing something and forgot to remove :) Tests are failing due to this

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**